### PR TITLE
feat: interactive docs using Pitsby

### LINKED ISSUE #26: Add different examples
Related https://github.com/theopensystemslab/map/pull/25

### PR DESCRIPTION
**Pull request by jessicamcinchak**
trying out [Pitsby](https://pitsby.com/) as a much lighter alternative to storybook that supports vanilla web components (Closes #26)

deployed here: https://deploy-preview-61--oslmap.netlify.app/#!/components/vanilla

**get up & running locally:**
- checkout branch, `pnpm i` to add new pitsby dependency
- `pnpm run docs` builds the docs in watch mode & listens for changes; site is running at [https://localhost:7000](https://localhost:7000)
- `pnpm docsPublish` builds the docs so they can be served as a static site from `pitsby/`

**deploying the docs:**
I've updated Netlify "deploy settings" to publish the component library instead of the `index.html` sandbox (that's still around for local dev though). I additionally removed our OS API keys from Netlify, otherwise Pitsby would read them & put them in plaintext in the "template" section. These docs are a bring-your-own-API-key situation! 

https://app.netlify.com/sites/oslmap/settings/deploys

before:
Screenshot from 2022-08-05 12-47-18

after:
Screenshot from 2022-08-05 12-55-05

**future cleanup:** 
- build out more examples - custom styles using css overrides, more recent map properties, etc
- what's happening with os features examples? why is data synced?
- geojsonData example not loading?
- some way to handle env vars / OS API keys? 
- typescript support?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,7 +19,9 @@
     "dev": "vite",
     "build": "rm -rf dist types && tsc && vite build && sed 's/src=\".*\"/src=\"component-lib.es.js\"/' index.html > dist/index.html",
     "prepublishOnly": "npm run build",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "docs": "NODE_ENV=development pitsby build --watch",
+    "docsPublish": "pitsby build"
```
Comment (DafyddLlyr) [inline package.json:24]: 👍

**Review by DafyddLlyr — APPROVED** [review]
This is great, and super nice to use! A big improvement and a solid step forward.

The future cleanup section of your PR was great - I hit the same issues but totally think that's a next step to look at as opposed to doing it all here.

File: pitsby/index.html [hunk 2]
```diff
@@ -0,0 +1 @@
+<!DOCTYPE html><html><head><meta charset="utf-8"><title>Docs - Place Components</title><meta name="viewport" content="width=device-width,initial-scale=1,user-scalable=yes"><meta http-equiv="cache-control" content="no-cache"><meta http-equiv="cache-control" content="max-age=0"><meta http-equiv="expires" content="0"><meta http-equiv="expires" content="Tue, 01 Jan 1980 1:00:00 GMT"><meta http-equiv="pragma" content="no-cache"><link href="images/favicon-pitsby.png?t=1659696039325" rel="shortcut icon"><link href="external/dist/style.css?t=1659696039325" rel="stylesheet"><script defer="defer" src="main.js"></script><link href="main.css" rel="stylesheet"></head><body ng-app="pitsby-app"><ui-view></ui-view><script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.8.3/angular.js"></script><script src="external/dist/component-lib.es.js?t=1659696039325"></script><script src="external/dist/component-lib.umd.js?t=1659696039325"></script></body></html>
```
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply pitsby/index.html:1]: absolutely, good call on this!

File: src/components/address-autocomplete/address-autocomplete.doc.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
+module.exports = {
+  name: "AddressAutocomplete",
+  description:
+    "AddressAutocomplete is a Lit wrapper for the Gov.UK accessible-autocomplete component that fetches & displays addresses in a given postcode using the Ordnance Survey Places API.",
+  properties: [
```
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply src/components/address-autocomplete/address-autocomplete.doc.js:5]: Yep totally agree - maybe something we can hack going forward to help make this more future proof - agree it's a bit tedious now.

File: package.json [hunk 4]
```diff
@@ -19,7 +19,9 @@
     "dev": "vite",
     "build": "rm -rf dist types && tsc && vite build && sed 's/src=\".*\"/src=\"component-lib.es.js\"/' index.html > dist/index.html",
     "prepublishOnly": "npm run build",
-    "prepare": "husky install"
+    "prepare": "husky install",
+    "docs": "NODE_ENV=development pitsby build --watch",
```
[the remark this answers is not in the document]
Comment (jessicamcinchak) [reply package.json]: yep appreciate this - will now run on 7007 :+1:

**Pull request merged**